app/src/components/UnitForm/index.js

```diff
@@ -1,188 +1,353 @@
     { value: 8, label: "8 months" },
     { value: 12, label: "12 months" },
   ];
 
+  const laundryOptions = ["Building", "Ensuite", "None"];
+  const typeOptions = ["Apartment", "House"];
+
   const defaultUnitValues = {
-    address: "301 Phillip St",
-    room: 3,
-    price: 500,
-    numBeds: 5,
-    numWashrooms: 5,
-    leaseDuration: 4,
-    floor: 3,
-    selectedImage: "",
+    address: "",
+    room_num: "",
+    rent_price: 500,
+    num_beds: 5,
+    num_washrooms: 5,
+    lease_term: 4,
+    floor_num: "",
+    image_path: "",
+    new_address: "",
+    pet_friendly: 0,
+    laundry_availability: "Building",
+    type_of_unit: "Apartment",
+    distance_from_waterloo: "0.0",
+  };
+
+  const defaultBuildingValues = {
+    new_address: "",
+    pet_friendly: 0,
+    laundry_availability: "Building",
+    type_of_unit: "Apartment",
+    distance_from_waterloo: "0.0",
   };
 
   const [postData, setPostData] = useState(defaultUnitValues);
+  const [newbuilding, setNewBuilding] = useState(defaultBuildingValues);
+  const [checked, setChecked] = useState(false);
+  const [message, setMessage] = useState("");
 
   const handleNum = (e) => {
     let value = parseInt(e.target.value, 10);
     if (value < 0) value = 0;
-    handleChange(e);
+    setPostData({
+      ...postData,
+      [e.target.name]: value,
+    });
   };
 
-  const handleChange = (e) => {
+  const handleAddress = (e) => {
+    const addrId = addressDict[e.target.value];
     setPostData({
       ...postData,
       [e.target.name]: e.target.value,
+      building_id: addrId,
     });
   };
-  const handleSubmit = async (e) => {
-    e.preventDefault();
-    const res = await axios.post("/api/unit/create", {
+
+  const handleBuilding = (e) => {
+    setNewBuilding({
+      ...newbuilding,
+      [e.target.name]: e.target.value,
+    });
+  };
+
+  const handleCheck = (e) => {
+    setChecked(e.target.checked);
+    const val = checked ? 0 : 1;
+    setNewBuilding({
+      ...newbuilding,
+      [e.target.name]: val,
+    });
+  };
+
+  const handleChange = (e) => {
+    setPostData({
       ...postData,
+      [e.target.name]: e.target.value,
     });
-    setPostData(defaultUnitValues);
+  };
 
-    handleClose();
+  const handleSubmit = async (e) => {
+    e.preventDefault();
+    if (postData.address === "Other") {
+      setPostData({
+        ...postData,
+        address: newbuilding.new_address,
+        room_num: parseInt(postData.room_num, 10),
+      });
+      setNewBuilding({
+        ...newbuilding,
+        distance_from_waterloo: parseFloat(newbuilding.distance_from_waterloo),
+      });
+      Object.keys(newbuilding).forEach((key) => {
+        postData[key] = newbuilding[key];
+      });
+    }
+    try {
+      const { data } = await axios.post("/api/unit/create", {
+        ...postData,
+      });
+      if (data.success) {
+        setPostData(defaultUnitValues);
+        handleClose();
+        navigate(`/unit/${data.unit_id}`);
```

Comment:
Question: After creating a new unit, I'm redirected to the /units/10 page, but it's blank. Is this the intended behaviour?
![Screenshot 2022-11-16 at 10 28 43 PM](https://user-images.githubusercontent.com/18638226/202347875-ff2764e0-851d-4664-9455-4fa1e5d7b4e3.png)

Reply:
yeah, I think the suggested behavior is to take the user to the newly created unit detail page, but we didn't implement the page yet, so it's blank right now.